internal/ir/xds.go

```diff
@@ -470,6 +481,15 @@ func (r DirectResponse) Validate() error {
 	return errs
 }
 
+// Direct response holds the details for mirroring requests to a second cluster
+// +k8s:deepcopy-gen=true
+type RouteMirror struct {
+	// The name of the route mirror is used to create a new cluster
+	ClusterName string
```

Comment:
is the `ClusterName` field needed ? can we create a suffixed `-mirror` cluster instead ?

Reply:
if this is reg optimization, we can optimize cluster creation in https://github.com/envoyproxy/gateway/issues/551

Reply:
Yeah, I'm cool with getting rid of it and just doing `routename-mirror-0`/`routename-mirror-1` based on the order they appear.

Reply:
sg, thanks @AliceProxy !